gobblin-core/src/main/java/org/apache/gobblin/publisher/BaseDataPublisher.java

```diff
@@ -290,6 +290,7 @@ public void close()
       for (Path path : this.publisherOutputDirs) {
         this.state.appendToSetProp(ConfigurationKeys.PUBLISHER_DIRS, path.toString());
       }
+      this.state.setProp(ConfigurationKeys.PUBLISHER_LATEST_FILE_ARRIVAL_TIMESTAMP, System.currentTimeMillis());
```

Comment:
I would suggest make `System.currentTimeMillis()` into a protected method as the default implementation(with a state object as parameter input) so that it can be easily extended in the future.

Reply:
Requirement changed to add a flag called update_operation=new_files_added. should that be added here or in gobblin-kafka?